### ISSUE TITLE
Remove disabled form element states

### ISSUE THREAD

**Issue by caw310** [department-of-veterans-affairs/vets-design-system-documentation#751]
There have been a few requests due to a11y issues to remove disabled element states from forms. The following tasks are to address those requests. 

- [ ] Draft guidance (work with Josh or Angela) to explain why we don't and shouldn't have disabled states 
- [ ] Add the guidance to the forms pattern documentation 
- [ ] Add the guidance to component documentation

**Comment by caw310** [comment]
@SarahKay8 , would you work with Courtney on this please when you have time? Thanks!

**Comment by k80bowman** [comment]
@CourtneyChambers77 thanks for considerations. We’ve made the decision to not have disabled states already, what we need is some guidance written for why we don’t have disabled states on form elements that we can point to when people ask for them. Did you reach out to Josh or Angela about this? Did they give you any advice?

**Comment by CourtneyChambers77** [comment]
Got it working on this

**Comment by k80bowman** [comment]
@SarahKay8 would you be able to work with Courtney on this?

**Comment by SarahKay8** [comment]
> @SarahKay8 would you be able to work with Courtney on this?

Will do!

**Comment by SarahKay8** [comment]
@k80bowman @caw310 hi team. Could you provide expected turn around time for this ticket? Wrapping up some midpoint work

**Comment by caw310** [comment]
@SarahKay8, this isn't urgent. Would be good if you have time to help with this either this week or next week. Thanks!

**Comment by SarahKay8** [comment]
@k80bowman I made some edits. This should show a user why disabled form elements as a whole wouldn't be a suitable option.

**Comment by CourtneyChambers77** [comment]
@SarahKay8  SarahKay8 @k80bowman I wanted to add in some bullets please Sarah and let me Know: Disabled button design guidance 
•	When the designers indicate an inactive state by graying out the button, the grey buttons with gray labels fails and does not meet color contrast recommendations for text.
•	The buttons are not focusable and can’t reach them when using a keyboard.
•	Show well-designed error message on button click.

**Comment by GnatalieH** [comment]
@CourtneyChambers77 I think @SarahKay8 covered these points in her write up, she just used different verbiage. But these themes -- insufficient contrast, lack of focus, and the importance of giving the user meaningful feedback are covered, I think.

Disabled button design guidance 
• When the designers indicate an inactive state by graying out the button, the grey buttons with gray labels fails and does not meet color contrast recommendations for text. 
• The buttons are not focusable and can’t reach them when using a keyboard. 
• Show well-designed error message on button click.

**Comment by k80bowman** [comment]
Holding off on saying that we will disable the form element states. @CourtneyChambers77 and @SarahKay8 do you have thoughts on what people could use instead? We would like to add this to the guidance.

**Comment by k80bowman** [comment]
I created a ticket to audit the use of disabled form states, to give us some more information #830

**Comment by SarahKay8** [comment]
@k80bowman My opinion is just for button but maybe it can be applied to other form elements. I've read different opinions about disabled/enabled button. My take is: Using disabled/enabled state on button has more cons
1. Disabled button will usually cause <= 4.5:1 contrast ratio for the button
2. Disabled button will be bypassed by screen reader so users will not be aware of the existing button
My preferred approach
1. Make sure to mark any required elements as such - using aria-required or accessible text. With these, we can mitigate the chance of premature form submission.
2. Do not put the button on disabled state. If user click it without entering all required attributes, announce the error (and perhaps shift focus to first unfilled required form element)
Just my 2 cents 🙂

**Comment by andresriveratoro** [comment]
Please update https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/707 and https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/896 when a decision has been reached.

**Comment by andresriveratoro** [comment]
Please update `va-button` if a decision is made to disallow `disabled`.

**Comment by humancompanion-usds** [comment]
@andresriveratoro - I went to document the disabled button and noticed that we don't have an example in Storybook now for va-button AND when I apply the disabled attribute on va-primary-button, nothing happens. Do you need a design for the disabled state? 

Here are some guidelines:
* The disabled contrast ratio does not matter and thus the gray we use now is adequate. I've confirmed this with DK.
* An aria-label needs to be present as well as the disabled attribute on the disabled button.
* The hover state should change the cursor from the typical hand to a symbol indicating that the button (or form field) cannot be interacted with. The CSS code is ```cursor: not-allowed;```

Confirmation from DK:
> Re: whether or not disabled buttons need to meet color contrast - Right, they don’t need to meet color contrast requirements. And if we’re following all the best practices you mention, use of disabled attribute, cursor change, proper labeling, we should be good.

The reason for this is here: https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html
> Incidental: Text or images of text that are part of an inactive [user interface component(https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html#user-interface-componentdef), that are [pure decoration](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html#puredecdef), that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement.

That essentially means that disabled buttons, being inactive, do not need to meet contrast requirements. That said, color should not be the only way to determine if an element is disabled, which is why I'm asking us to add those other changes. Also, we're going to continue to strongly discourage folks from disabling buttons and form states. Buttons is the start of that broader guidance.

**Comment by humancompanion-usds** [comment]
@k80bowman - Can I get a quick audit of which form fields currently are able to be disabled via a property on the component? I'm not actually finding any. Also, does the lack of a property on the component mean that it cannot be disabled? Because it sure looks like they are many instances of components being disabled on va.gov. I guess what I'm asking is: Is this issue even achievable? Or could folks just add "disabled" to any form field manually anyway or disable in another, non-approved way?

**Comment by caw310** [comment]
Other related tickets: 
https://app.zenhub.com/workspaces/vsp-design-system-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/830

https://app.zenhub.com/workspaces/vsp-design-system-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/343

https://app.zenhub.com/workspaces/vsp-design-system-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/342

https://app.zenhub.com/workspaces/vsp-design-system-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/707

https://app.zenhub.com/workspaces/vsp-design-system-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/896

**Comment by humancompanion-usds** [comment]
We've now published guidance recommending that teams do NOT disable buttons. That's in 2 places:

https://design.va.gov/components/button/#do-not-disable-buttons
https://design.va.gov/components/form/#accessibility-considerations-for-all-form-elements

We'll be adding more use cases and recommendations to that 2nd link in the upcoming sprint. We're now waiting though while the audit of instances is completed.
